### ISSUE TITLE
Setup instructions with Channels 3

### ISSUE THREAD

**Issue by foucdeg** [fanout/django-eventstream#59]
Channels 3 is out and its [migration guide](https://channels.readthedocs.io/en/stable/releases/3.0.0.html) warns us of a few breaking changes and deprecated classes, which the setup instructions here use.

Is this lib compatible with Channels 3?
Is the example code different?

**Comment by jkarneges** [comment]
I think some changes would need to be made to support Channels 3. Will look at this soon.

**Comment by foucdeg** [comment]
Great, thanks!

**Comment by foucdeg** [comment]
Hey @jkarneges , any updates ? :)

**Comment by jkarneges** [comment]
I'm considering moving to Django 3 which has built-in support for async views. Perhaps we don't need Channels anymore. I still need to plan this out though.

**Comment by foucdeg** [comment]
OK, I'm also using Django 3 on my project. I'd be interested in any setup instructions without Channels. No dependency is even better than an up to date dependency :)

**Comment by bertrand-benoit** [comment]
Hello, I'm interested too.
My current attempt: Python 3.8, Django 3.1.4, Django-eventstream 3.1.0, is failing when trying to serve any page with something like:
```
[...]
File "/home/bsquare/.local/share/virtualenvs/lynceus_django_just_created-f6Z0-ibI/lib/python3.8/site-packages/asgiref/compatibility.py", line 33, in new_application
    instance = application(scope)
TypeError: __init__() takes 1 positional argument but 2 were given
```

I guess it is related to what reported @foucdeg about migration ... till then, is there any way to get a workaround now?

**Comment by hadpro24** [comment]
> Hello, I'm interested too.
> My current attempt: Python 3.8, Django 3.1.4, Django-eventstream 3.1.0, is failing when trying to serve any page with something like:
> 
> ```
> [...]
> File "/home/bsquare/.local/share/virtualenvs/lynceus_django_just_created-f6Z0-ibI/lib/python3.8/site-packages/asgiref/compatibility.py", line 33, in new_application
>     instance = application(scope)
> TypeError: __init__() takes 1 positional argument but 2 were given
> ```
> 
> I guess it is related to what reported @foucdeg about migration ... till then, is there any way to get a workaround now?

Use channels 2.4.0 fix it

**Comment by bertrand-benoit** [comment]
> > Hello, I'm interested too.
> > My current attempt: Python 3.8, Django 3.1.4, Django-eventstream 3.1.0, is failing when trying to serve any page with something like:
> > ```
> > [...]
> > File "/home/bsquare/.local/share/virtualenvs/lynceus_django_just_created-f6Z0-ibI/lib/python3.8/site-packages/asgiref/compatibility.py", line 33, in new_application
> >     instance = application(scope)
> > TypeError: __init__() takes 1 positional argument but 2 were given
> > ```
> > 
> > 
> > I guess it is related to what reported @foucdeg about migration ... till then, is there any way to get a workaround now?
> 
> Use channels 2.4.0 fix it
Hello, thanks for your answer ... but what are you speaking about?
Atm, we use django-channels, embedded in Django 3.x version.

And I do not see any v.4.x on channels? See. https://github.com/django/channels/releases

**Comment by hadpro24** [comment]
> > > Hello, I'm interested too.
> > > My current attempt: Python 3.8, Django 3.1.4, Django-eventstream 3.1.0, is failing when trying to serve any page with something like:
> > > ```
> > > [...]
> > > File "/home/bsquare/.local/share/virtualenvs/lynceus_django_just_created-f6Z0-ibI/lib/python3.8/site-packages/asgiref/compatibility.py", line 33, in new_application
> > >     instance = application(scope)
> > > TypeError: __init__() takes 1 positional argument but 2 were given
> > > ```
> > > 
> > > 
> > > I guess it is related to what reported @foucdeg about migration ... till then, is there any way to get a workaround now?
> > 
> > 
> > Use channels 2.4.0 fix it
> 
> Hello, thanks for your answer ... but what are you speaking about?
> Atm, we use django-channels, embedded in Django 3.x version.
> 
> And I do not see any v.4.x on channels? See. https://github.com/django/channels/releases

No use the : v2.1.0

**Comment by bertrand-benoit** [comment]
> No use the : v2.4.0

So channels v2.4.0 can be used with Django 3.1.x ?

**Comment by foucdeg** [comment]
@bertrand-benoit yes, it works on my app. I opened this issue about Channels 3 + Django 3.

**Comment by hadpro24** [comment]
> > No use the : v2.4.0
> 
> So channels v2.4.0 can be used with Django 3.1.x ?

Yes try it

**Comment by foucdeg** [comment]
Hey @jkarneges , any news on this ? :)

**Comment by jkarneges** [comment]
I'm planning to start looking at this before the end of the month.

**Comment by jkarneges** [comment]
It seems Django [does not yet support](https://code.djangoproject.com/ticket/31852) async streaming responses. We'll need Channels for a bit longer then.

Will update again soon.

**Comment by jkarneges** [comment]
Released 4.0.0 with support for Channels 3.

**Comment by foucdeg** [comment]
Awesome, thanks! Since both my local and production environments use Pushpin, I was able to remove Channels altogether.
